src/coreclr/tools/Common/JitInterface/CorInfoImpl.cs

```diff
@@ -877,7 +907,7 @@ private void getMethodSig(CORINFO_METHOD_STRUCT_* ftn, CORINFO_SIG_INFO* sig, CO
                 }
             }
 
-            Get_CORINFO_SIG_INFO(method, sig);
+            Get_CORINFO_SIG_INFO(method, _compilation.GetMethodIL(method), sig: sig);
```

Comment:
Getting the method IL every time we get any method signature feels expensive. 

Since the suppress GC transition for calli is a hack that will go away, it maybe better to keep it as local hack:
- Keep TODO: Replace this with a public mechanism to mark calli with SuppressGCTransition once it becomes available
- Set a locally defined CORINFO_SIGFLAG_SUPPRESS_GC_TRANSITION flag in `sig->flags` 
- Check this flag in `getUnmanagedCallConv`